### ISSUE TITLE
Log container init when max verbosity is set

### ISSUE THREAD

**Issue by maxgio92** [kubernetes-sigs/kind#3575]
**What would you like to be added**:

Hello, I propose to log all the container logs from before init when setting the maximum verbosity supported.

**Why is this needed**:

This is extremely helpful to debug at runtime with logs from init and before, without the need to read logs afterward with `export logs`.

A lot of points can go wrong e.g. when setting up cgroups and in that case logs from Systemd are essential.
Combining them with logs from kind itself first and from Kubernetes then in a single output can be vital to better understand at runtime the initialization sequence of the containerized nodes, and possibly debug and resolve issues.

As of now only the log of the multi-user systemd target reached event is filtered, leading to ambiguous outputs like [this one](https://github.com/kubernetes-sigs/kind/issues/3423#issue-1992335390).

**Comment by aojea** [comment]
naive question, how this will solve this issue https://github.com/kubernetes-sigs/kind/issues/3423#issue-1992335390 ?
what different will this make?

**Comment by maxgio92** [comment]
Hi @aojea, I didn't mean that it solves that issue.
I meant that there are cases where init logs at runtime, like the case related to that issue, are useful for debug.

**Comment by aojea** [comment]
sure, just  trying to understand what are those cases, on my experience kind export logs is always enough, so I'm trying to understand what I'm missing

**Comment by maxgio92** [comment]
I get what you mean, I agree that the same logs are available also with export logs, and at the end you can use both ways to debug, but in this case without needing to save all available logs on disk.
I'm not saying that one way is better than another but that sometimes you don't need all the logs to be exported and just the init log at runtime can provide enough information to debug issues like the one linked.

**Comment by BenTheElder** [comment]
Usually the container log isn't where the problem is, and we wind up needing the component logs.

**Comment by maxgio92** [comment]
That's why I'm proposing to add the ability to make it print them (eg systemd logs) when maximum verbosity is set

**Comment by BenTheElder** [comment]
> That's why I'm proposing to add the ability to make it print them (eg systemd logs) when maximum verbosity is set, besides the ability to export them all

I don't see why this makes more sense than export logs, it will be really hard to pick through _all_ of these being dumped from the same process and I've never seen another cluster tool do this.

**Comment by BenTheElder** [comment]
I think this is just going to cause users to paste useless log snippets and ignore the instructions to use `kind export logs` for detailed logs when things break.

What if we automatically did ~`kind export logs` on `kind create cluster` failure, with some knobs to opt out and to control where these go?

**Comment by maxgio92** [comment]
I agree with the proposal @BenTheElder. It would probably help reading that errors have been exported in case of failure, to have more details about the cause of it.
